oscar_elasticsearch/search/indexing/indexer.py

```diff
@@ -129,7 +129,9 @@ def reindex(self):
 
         try:
             yield self
-        finally:
+        except Exception:  # pylint: disable=try-except-raise
```

Comment:
In stead of:
```
try:
    yield self
except Exception:
    raise
else:
    self.indexer.finish()
```
I think you can just write it like this:
```
yield self
self.indexer.finish()
```
But I think you have to consult with viggo why the finish was there in the finally block first.

Reply:
But I think you have to consult with viggo why the finish was there in the finally block first.
- I did this some time ago, but didn't think of errors when I made this. So the new logic is correct.